File: JULY-2025/05-07-2025/p1394.kt
class Solution {
    fun findLucky(arr: IntArray): Int {
        val freq = IntArray(501)
        for (num in arr) {
            freq[num]++
        }

        var result = -1
        for (num in arr) {
            if (freq[num] == num) {
                result = maxOf(result, num)
            }
        }
        return result
    }
}

File: JULY-2025/05-07-2025/p1394.py
class Solution:
    def findLucky(self, arr: List[int]) -> int:
        freq = {}
        lucky = -1
        for num in arr:
            if num in freq:
                freq[num] += 1
            else:
                freq[num] = 1
        for i in freq:
            if freq[i] == i:
                lucky = max(lucky,i)
        return lucky